src/config/config.rs

```diff
@@ -17,37 +17,52 @@ use std::{
     str::FromStr,
 };
 
-fn canonicalize(path: &Path) -> String {
-    let mut parts: Vec<String> = Vec::new();
+fn relative_canonicalize(path: impl AsRef<Path>) -> PathBuf {
+    let mut parts: Vec<Component<'_>> = vec![];
 
-    for component in path.components() {
+    for component in path.as_ref().components() {
         match component {
-            Component::RootDir | Component::CurDir => {}
+            // :shrug:
+            // Component::RootDir | Component::CurDir => {}
             Component::ParentDir => {
-                parts.pop();
+                parts
+                    .pop()
+                    .expect("Cannot navigate outside of base directory!");
             }
-            c => parts.push(
-                c.as_os_str()
-                    .to_str()
-                    .expect("OsStr::to_str failed")
-                    .to_owned(),
-            ),
+            c => parts.push(c),
         }
     }
-    parts.join("/")
+    parts.iter().collect()
+}
+
+#[test]
+fn test_relative_canonicalize() {
+    assert_eq!(relative_canonicalize("foo/../bar").as_os_str(), "bar");
+    assert_eq!(
+        relative_canonicalize("foo/../bar/baz/../bav").as_os_str(),
+        "bar/bav"
+    );
+    assert_eq!(
+        relative_canonicalize("foo/bar/baz/../../bav").as_os_str(),
+        "foo/bav"
+    );
+    // Losing root...
+    assert_eq!(relative_canonicalize("/foo").as_os_str(), "foo");
```

Comment:
How come we lose the root with this implementation?

Reply:
Not anymore, because I commented it out.